### PR TITLE
Add 4.x modules and review editor/non-editor ones

### PR DESCRIPTION
**Pull request by DeeJayLSP**
Modules that belong in a certain version have a note.

I dived into the code to check what each module does.

Even if some listed modules are only useful in the editor, they are still compiled on non-editor builds, therefore they should be on the list.

If I did any mistake, let me know. I'm a bit confused about 4.x `raycast` since it is *de facto* editor only (doesn't seem to add anything usable to non-editor builds).

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -213,14 +228,28 @@ <h2>Resulting <code>custom.py</code></h2>
...
           },
-          // 'lightmapper_rd' isn't included as it's editor-only.
+          // 'lightmapper_cpu' (3.x only) isn't included as it's editor-only.
+          {
+            id: 'lightmapper_rd',
+            name: 'LightmapperRD',
+            description: 'Required to use LightmapperRD (4.x only).',
+            enabled: true,
+            default: true,
+          },
```
Comment (Calinou) [inline index.html]: I thought LightmapperRD was editor-only. To my knowledge, Godot doesn't support baking lightmaps at run-time at all, and I don't remember if UV2 wrapping is still exposed.

It's certainly a viable usecase in some scenarios now that lightmapping is GPU-accelerated, but you'd need a way to cache the UV2 and lightmaps to make subsequent map loads faster.

If it's not intended to be a supported usecase, we should make sure LightmapperRD isn't built in export templates. I've asked on the Godot Contributors Chat and will see what others have to say about it.

File: index.html [hunk 2]
```diff
@@ -279,15 +342,41 @@ <h2>Resulting <code>custom.py</code></h2>
...
             default: true,
           },
-          // 'squish' isn't included as it's editor-only.
+          {
+            id: 'squish',
+            name: 'Squish',
+            description: 'Required to decompress images compressed with Squish.',
+            enabled: true,
+            default: true,
+          },
```
Comment (Calinou) [inline index.html]:
```suggestion
          {
            id: 'squish',
            name: 'Squish',
            description: 'Required to decompress images compressed with Squish (including S3TC VRAM compression).',
            enabled: true,
            default: true,
          },
```

File: index.html [hunk 3]
```diff
@@ -279,15 +342,41 @@ <h2>Resulting <code>custom.py</code></h2>
...
+            id: 'svg',
+            name: 'SVG',
+            description: 'Required to load non-imported SVG images as raster.',
+            enabled: true,
+            default: true,
+          },
+          {
+            id: 'text_server_adv',
+            name: 'Text Server Advanced',
+            description: 'Required to use TextServerAdvanced (4.x only).',
```
Comment (Calinou) [inline index.html]:
```suggestion
            description: 'Required to use TextServerAdvanced, which is the default TextServer (4.x only).',
```

**Review by Calinou — APPROVED** [review]

**Comment by Calinou** [comment]
Should be good to merge after applying suggestions.

**Comment by DeeJayLSP** [comment]
Done

**Pull request merged**